File: frontend/data/models/customer.js
class Customer {
    constructor(customerID, latitude, longitude) {
        this.customerID = customerID;
        this.latitude = latitude;
        this.longitude = longitude;
    }

    static fromJSON(json) {
        if (!json || typeof json !== 'object' ||
            !json.hasOwnProperty('customerID') ||
            !json.hasOwnProperty('latitude') ||
            !json.hasOwnProperty('longitude')) {
            throw new Error('Invalid JSON data for Customer');
        }

        return new Customer(
            json.customerID,
            json.latitude,
            json.longitude
        );
    }
}

File: backend/api/Models/Customer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.Data.SqlClient;
using MySql.Data.MySqlClient;
using api.Models;
using api.Interfaces;

namespace api.Models
{
    public class Customer
    {
      public string CustomerId { get; set; } = string.Empty;
    
      public double Latitude { get; set; }

      public double Longitude { get; set; }

      public List<Order> Orders { get; set; } = new List<Order>();

    }
}